plugins/woocommerce/tests/e2e-pw/tests/merchant/create-simple-product.spec.js

```diff
@@ -83,13 +83,22 @@ test.describe( 'Add New Simple Product Page', () => {
 		await expect( page.locator( 'div.notice-success > p' ) ).toContainText(
 			'Product published.'
 		);
+
+		virtualProductPermalink = await page
+			.locator( '#sample-permalink a' )
+			.getAttribute( 'href' );
 	} );
 
 	test( 'can have a shopper add the simple virtual product to the cart', async ( {
 		page,
 	} ) => {
-		await page.goto( 'shop/' );
-		await page.click( `h2:has-text("${ virtualProductName }")` );
+		await page.goto( virtualProductPermalink );
+		await expect( page.locator( '.product_title' ) ).toHaveText(
+			virtualProductName
+		);
+		await expect(
+			page.locator( '.summary .woocommerce-Price-amount' )
+		).toContainText( productPrice );
```

Comment:
I noticed that adding the product from the `shop` page is flaky because:
1. The smoke test site already has 200+ products, which I believe are leftovers from previous tests, and sometimes the shop page shows them instead of the virtual/non-virtual product that this test created.
2. When this test runs in parallel with any of the "import products" tests, the `shop` page will be filled with those imported products, thereby pushing the virtual and non-virtual products in this test out. 

To address this flakiness, I modified the test to navigate to the product page instead using its permalink, and add it to the cart from there.